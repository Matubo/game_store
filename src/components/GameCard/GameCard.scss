.game-card {
  display: flex;
  flex-direction: column;
  align-items: center;
}

.game-card__img {
  height: 50px;
  width: 50px;
}
