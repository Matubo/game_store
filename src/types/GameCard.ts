export interface IGameCard {
  id: number;
  name: string;
  image: string;
  description?: string;
  rating?: number;
  ageLimit: number;
  price: number;
  platform: {
    pc: boolean;
    xbox: boolean;
    playstation: boolean;
  };
}
