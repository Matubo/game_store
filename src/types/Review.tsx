type IPlatform = { pc: boolean; playstation: boolean; xbox: boolean };

export type IReview = {
  id: number;
  name: string;
  rating: 4;
  genre: string;
  platform: IPlatform;
  image: string;
  description: string;
  alt: string;
};
