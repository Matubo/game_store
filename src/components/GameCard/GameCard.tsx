import { IGameCard } from 'src/types/gameCard';

import './GameCard.scss';

interface IProps {
  game: IGameCard;
  callback: (elem: IGameCard) => void;
}

export default function GameCard({ game, callback }: IProps) {
  const { image, name, price, description, rating, discount } = game;

  const clickHandler = (e: React.MouseEvent) => {
    e.preventDefault();
    callback(game);
  };

  return (
    <div className="search-result__game-card game-card">
      <div className="game-card__front">
        <div className="game-card__rating">{rating}/5&#9733;</div>
        {discount ? <div className="game-card__discount">{discount}%</div> : <></>}
        <img className="front__img" src={image} alt="game-pic"></img>
        <p className="front__name">{name}</p>
      </div>
      <div className="game-card__back">
        <p className="back__description">{description}</p>
        <div className="back__add-panel">
          <p className="add-panel__price">{price} â‚¬</p>
          <button className="add-panel__add-button" onClick={clickHandler}>
            +
          </button>
        </div>
      </div>
    </div>
  );
}
