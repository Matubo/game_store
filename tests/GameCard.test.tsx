import { fireEvent, render, screen } from '@testing-library/react';
import GameCard from 'src/components/GameCard/GameCard';
import { IGameCard } from 'src/types/gameCard';

describe('GameCard tests', () => {
  const gameExample: IGameCard = {
    ageLimit: 18,
    id: 1,
    image: '/srcTest',
    name: 'TestGame',
    platform: { pc: true, playstation: false, xbox: true },
    price: 1800,
    description: 'The Greatest test game for test',
    discount: 5,
    rating: 4
  };

  const callbackMock = jest.fn((game: IGameCard) => {
    return game;
  });

  const { getByRole, getByAltText } = render(<GameCard callback={callbackMock} game={gameExample}></GameCard>);
  const button = getByRole('button');
  fireEvent.click(button);

  test('button call handler', () => {
    expect(callbackMock).toBeCalledTimes(1);
  });

  test('callback return data', () => {
    expect(callbackMock.mock.results[0].value).toBe(gameExample);
  });

  test('img set url from prop', () => {
    render(<GameCard callback={callbackMock} game={gameExample}></GameCard>);
    const img = getByAltText('game-pic');
    expect((img as HTMLImageElement).src).toContain(gameExample.image);
  });

  test('img error handler', () => {
    render(<GameCard callback={callbackMock} game={gameExample}></GameCard>);
    const img = getByAltText('game-pic');
    expect((img as HTMLImageElement).src).not.toBe(gameExample.image);
  });

  test('snapshot test', () => {
    const { asFragment } = render(<GameCard callback={callbackMock} game={gameExample}></GameCard>);
    expect(asFragment()).toMatchSnapshot();
  });
});
