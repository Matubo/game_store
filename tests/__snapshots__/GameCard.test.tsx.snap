// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameCard tests snapshot test 1`] = `
<DocumentFragment>
  <div
    class="search-result__game-card game-card"
  >
    <div
      class="game-card__front"
    >
      <div
        class="game-card__rating"
      >
        4/5★
      </div>
      <div
        class="game-card__discount"
      >
        5%
      </div>
      <img
        alt="game-pic"
        class="front__img"
        src="/srcTest"
      />
      <p
        class="front__name"
      >
        TestGame
      </p>
    </div>
    <div
      class="game-card__back"
    >
      <p
        class="back__description"
      >
        The Greatest test game for test
      </p>
      <div
        class="back__add-panel"
      >
        <p
          class="add-panel__price"
        >
          1800 €
        </p>
        <button
          class="add-panel__add-button"
        >
          +
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
